pkg/costmodel/key.go

```diff
@@ -0,0 +1,348 @@
+package costmodel
+
+import (
+	"fmt"
+
+	"github.com/kubecost/cost-model/pkg/env"
+	"github.com/kubecost/cost-model/pkg/prom"
+)
+
+type containerKey struct {
+	Cluster   string
+	Namespace string
+	Pod       string
+	Container string
+}
+
+func (k containerKey) String() string {
+	return fmt.Sprintf("%s/%s/%s/%s", k.Cluster, k.Namespace, k.Pod, k.Container)
+}
+
+func newContainerKey(cluster, namespace, pod, container string) containerKey {
```

Comment:
Just looked nicer, syntactically, to my eye. Think we should just construct directly?

Reply:
I like constructing directly because it doesn't rely on ordered args and therefore makes it clearer what is being mapped to what.

Reply:
On the other hand, a `new...` function forces everything to be initialized. :man_shrugging:

Reply:
What do you think about this?
```go
podKey := podKey{cluster, namespace, pod}
```
as opposed to this:
```go
podKey := podKey{
	Cluster: cluster,
	Namespace: namespace,
	Pod: pod,
}
```

Pretty similar to `newContainerKey(cluster, namespace, pod)`, so we probably don't need that function, but I wonder if you have the same objection, then.

Reply:
To be honest, I think `podKey{cluster, namespace, pod}` is the worst of both worlds syntactically - it doesn't force parameters to be passed _and_ it isn't clear (imo) what is being mapped to what in the struct.

I like
```
podKey := podKey{
	Cluster: cluster,
	Namespace: namespace,
	Pod: pod,
}
```
the most, especially because `podKey` has no unexported/special fields that need to be set by a constructor.

I'm worried I'm diving way to far into nitpicking territory, so feel free to just ignore this. What you've got will work totally fine, I think I knee-jerked at an unexpected struct constructor.